quickstart/dotnet-core/Program.cs

```diff
@@ -207,102 +208,97 @@ public static async Task ForceReconnectAsync()
 
         // In real applications, consider using a framework such as
         // Polly to make it easier to customize the retry approach.
-        private static async Task<T> BasicRetryAsync<T>(Func<T> func)
+        private static async Task<T> BasicRetryAsync<T>(Func<IDatabase, Task<T>> func)
         {
             int reconnectRetry = 0;
-            int disposedRetry = 0;
 
+            IDatabase cache = Connection.GetDatabase();
             while (true)
             {
                 try
                 {
-                    return func();
+                    return await func(cache);
                 }
                 catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
                 {
                     reconnectRetry++;
                     if (reconnectRetry > RetryMaxAttempts)
                         throw;
-                    await ForceReconnectAsync();
-                }
-                catch (ObjectDisposedException)
-                {
-                    disposedRetry++;
-                    if (disposedRetry > RetryMaxAttempts)
-                        throw;
+                    try
+                    {
+                        await ForceReconnectAsync();
+                        cache = Connection.GetDatabase();
+                    }
+                    catch (ObjectDisposedException) { }
                 }
             }
         }
 
-        public static Task<IDatabase> GetDatabaseAsync()
+        static async Task Main(string[] args)
         {
-            return BasicRetryAsync(() => Connection.GetDatabase());
-        }
+            await InitializeAsync();
 
-        public static Task<System.Net.EndPoint[]> GetEndPointsAsync()
-        {
-            return BasicRetryAsync(() => Connection.GetEndPoints());
-        }
+            // Perform cache operations using the cache object...
+            Console.WriteLine("Running... Press 'q' to quit.");
+            ConsoleKey inputKey = ConsoleKey.A; // Init value, it may be anything different to Q
 
-        public static Task<IServer> GetServerAsync(string host, int port)
-        {
-            return BasicRetryAsync(() => Connection.GetServer(host, port));
-        }
+            while (inputKey != ConsoleKey.Q)
+            {
+                for (int i = 0; i < 2; i++)
+                {
+                    Task.Run(() => RunRedisCommands());
+                    Thread.Sleep(500);
+                }
 
-        static async Task Main(string[] args)
-        {
-            await InitializeAsync();
+                Thread.Sleep(2000);
 
-            IDatabase cache = await GetDatabaseAsync();
+                if (Console.KeyAvailable)
+                {
+                    inputKey = Console.ReadKey().Key;
+                }
+            }
```

Comment:
I think this can be simplified to: 
`
            Console.WriteLine("Running... Press any key to quit.");

            while (!Console.KeyAvailable)
            {
                Task thread1 = Task.Run(() => RunRedisCommandsAsync());
                Task thread2 = Task.Run(() => RunRedisCommandsAsync());

                Task.WaitAll(thread1, thread2);
            }
`

Reply:
Love it. My only thought is, making the example multi-threaded unfortunately messes up the clarity of the output, esp. without any wait in between. Do you think it would be better to go back to a single-threaded example, or maybe I should prepend each log statement with the threadId so that it's clear which command & response go together?

Reply:
We definitely want the threads to be running concurrently to demonstrate the thread-safety, so confusing interleaved log statements are actually good.
BUT we can clarify them by prefixing each log line with "Thread 1:" or "Thread 2:". You could add a param to `RunRedisCommands()` for that prefix